test/objects_test.js

```diff
@@ -49,4 +49,39 @@ describe('objects', () => {
     assert.deepEqual(Page.fromRaw({data: 1, before: 2, after: 3}), new Page(1, 2, 3))
     assert.deepEqual(new Page([1, 2, 3], 2, 3).mapData(_ => _ + 1), new Page([2, 3, 4], 2, 3))
   })
+
+  it('time conversion', () => {
+    const dt = new Date()
+    assert.deepEqual(new FaunaTime(dt).date, dt)
+
+    const epoch = new Date(Date.UTC(1970, 0, 1))
+    const ft = new FaunaTime(epoch)
+    assert.deepEqual(ft, new FaunaTime('1970-01-01T00:00:00.000Z'))
+    assert.deepEqual(ft.date, epoch)
+  })
+
+  it('time', () => {
+      const test_ts = new FaunaTime('1970-01-01T00:00:00.123456789Z')
```

Comment:
there's got to be an offset parser in javascript out there. its part of the ISO 8601 or whatever it is format. I'd be surprised if it wasn't part of the standard lib.

And its simple in this case, because its a simple offset; it does not contain logical timezone information, such as DST.